components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuCandidate.kt

```diff
@@ -159,6 +164,82 @@ data class ContextMenuCandidate(
             }
         )
 
+        /**
+         * Context Menu item: "Add to contact".
+         */
+        fun createAddContactCandidate(
+            context: Context
+        ) = ContextMenuCandidate(
+            id = "mozac.feature.contextmenu.add_to_contact",
+            label = context.getString(R.string.mozac_feature_contextmenu_add_to_contact),
+            showFor = { _, hitResult -> hitResult.isMailto() },
+            action = { _, hitResult ->
+                val link = hitResult.getLink()
+                val intent = Intent(ContactsContract.Intents.Insert.ACTION).apply {
+                    type = ContactsContract.RawContacts.CONTENT_TYPE
+                    putExtra(ContactsContract.Intents.Insert.EMAIL, link.stripMailToProtocol())
+                    putExtra(
+                        ContactsContract.Intents.Insert.EMAIL_TYPE,
+                        ContactsContract.CommonDataKinds.Email.TYPE_WORK
+                    )
+                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                }
+
+                context.startActivity(intent)
```

Comment:
This looks like a common functionality that other consumer could take advantage of, maybe we can put is in `Context` as extension function. What do you think?

Reply:
Sure, I'll look into moving this to a `Context` as an extension function.  Thanks,